### ISSUE TITLE
If a FT adds a Location, include the field parameters (as though they had been checked when setting up the task).

### ISSUE THREAD

**Issue by bsaiken** [FWAJL/FieldWorkManager#1207]

**Comment by gsouvik** [comment]
@bsaiken  
So when the FT adds a location, for every field data the task has, a corresponding entry has to be added to the table `field_analyte_location`. That way the `/task/fieldMatrix` when opened would have the checkboxes for that location + field data checked. Is my understanding on this correct?

**Comment by bsaiken** [comment]
@gsouvik 
Yes.  

We added the checkboxes so that a PM could select with location/parameter pair would need to be sampled.  The default should be everything is sampled, so it is more accurate to say the PM could de-select some pairs.  So, the normal condition should be "checked", and a row would be added to `field_analyte_location` for every pair unless the PM has taken action to remove the pair.  If the FT adds a location live in the field, the PM will not have the opportunity to de-select any pairs (this action is not allowed for an active task).  

That paragraph was more for me than you.  Just checking to make sure I am thinking straight.

**Comment by gsouvik** [comment]
@bsaiken 
I've coded the feature as per our discussion. When FT adds a new location, the field matrix for that field data + location combination is automatically created. Code is pushed to feature_devsupport.

**Comment by bsaiken** [comment]
@gsouvik 
I just tried this setting up a new Task and this did not work - boxes will not stay checked.  
In addition, I have enough field data parameter that it is spilling over onto page 2 and so the pagination issue must be fixed.  The error may be because of the pagination.  In any case, I cannot keep the boxes checked.
The Task can be accessed by using the login credentials: `elcampo` : `elcampo`  so you can see live what the issue is.

**Comment by gsouvik** [comment]
@bsaiken 
I just did a git pull and tested the requirement which was defined for this issue - when FT creates location, the `field_analyte_location` table is populated with the relation and FT can immediately add results for those location + field data combinations. 

I could not test the same on live (http://dev.fieldworkmanager.com) as the login was not a FT login. I opened the field matrix for El Campo > Telemitry > Field Parameters and saw just one Field data. I may be missing something. Please let me know if I am.

Let me spot the issue which addressed the multi page field matrix check box problem and fix it. As you suggested that may help fix this one too.

**Comment by FWAJL** [comment]
@gsouvik I recommend loading the complete matrix onto the view module and handle the paging using jQuery. The current problem is due to page load when changing page.

Can you do that?

**Comment by gsouvik** [comment]
@bsaiken 
Yes that could be done. But that would be nearly equal amount of code change to the fix where we actually keep things as it is and take care of the pagination check box issue on the php side.
It is generally a good idea to fix the root cause (which is how the matrix is saved on the php side) than to cover it up with a workaround. But I'm open to either way, your call.

**Comment by bsaiken** [comment]
@gsouvik 
The problem is created by pagination.  
Understand that as I am using the site I am finding usability issues which differ from my conceptual idea of how the page would work.  
Please make two changes to fix this view.
1) Remove pagination.  The PM will need to scroll to the right. 
Question - can the field parameter/location pairs be placed inside a scrollable div?  In other words, can you make only the area shown in the red box be scrollable, left to right?
<img width="1440" alt="screen shot 2015-08-25 at 8 37 11 am" src="https://cloud.githubusercontent.com/assets/8228624/9467981/c5746dcc-4b04-11e5-8359-85339389a4a0.png">

2) We need to display more of the text for location names and parameter headings.  I still want the tooltip for very large entries; however, make the changes you originally suggested here: https://github.com/FWAJL/FieldWorkManager/issues/958#issuecomment-99973664
This has two lines of text before we abbreviate. 

Also, make the following CSS changes:
.matrix-cell {
  margin: 1px;
}
.matrix-alalyte-header {
  font-size: 16px;
}
.matrix-location-header {
  font-size: 16px;
}
.matrix-cell-data {
  font-size: 12px;
}

**Comment by gsouvik** [comment]
@bsaiken 
Yes, I think what you are asking for could be done. Do want me to make the commit under #1148?

**Comment by gsouvik** [comment]
@bsaiken 
I think this is solved as part of #1148. Let me know if anything remains on this issue which needs to be looked into.

**Comment by bsaiken** [comment]
@gsouvik 
Yes, his is working.
